### ISSUE TITLE
[CI] re-try network requests / add jacocoTestReport progress output

### ISSUE THREAD

**Issue by mikehardy** [ankidroid/Anki-Android#5605]
###### Reproduction Steps

1. Attempt to use JDK11 in CI
2. see it fail with below stack
3. 

###### Expected Result

Should run - it was running after #5594 bit I apparently re-broke it in #5602 

Most likely due to factoring out the jacoco stuff.

If no one else gets to it I will definitely patch it up

###### Actual Result

```
The command "if [ "$UNIT_TEST" = "TRUE" ]; then ./gradlew jacocoUnitTestReport; fi" exited with 0.
20.05s$ if [ "$API" != "NONE" ]; then ./gradlew jacocoAndroidTestReport; fi
Welcome to Gradle 5.6.3!
Here are the highlights of this release:
 - Incremental Groovy compilation
 - Groovy compile avoidance
 - Test fixtures for Java projects
 - Manage plugin versions via settings script
For more details see https://docs.gradle.org/5.6.3/release-notes.html
Starting a Gradle Daemon (subsequent builds will be faster)
FAILURE: Build failed with an exception.
* Where:
Build file '/home/travis/build/ankidroid/Anki-Android/AnkiDroid/build.gradle' line: 8
* What went wrong:
A problem occurred evaluating project ':AnkiDroid'.
> Failed to apply plugin [id 'com.android.application']
   > Could not create plugin of type 'AppPlugin'.
      > Could not generate a decorated class for class com.android.build.gradle.AppPlugin.
         > com/android/builder/errors/EvalIssueReporter
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 19s
The command "if [ "$API" != "NONE" ]; then ./gradlew jacocoAndroidTestReport; fi" exited with 1.
```

###### Research
*Enter an [ x ] character to confirm the points below:*

[  ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[  ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[  ] I have searched for similar existing issues here and on the user forum

**Comment by mikehardy** [comment]
Also important to note that testDebugUnitTest still occasionally hangs (maybe turn on verbosegc or something?), and sdkmanager still does a connection reset sometimes. Maybe I need a retry-fetcher for that one, it has a proper exit error code to key off

**Comment by mikehardy** [comment]
The JDK="1.11" failure is difficult to diagnose. It only fails on the main AnkiDroid CI instance on travis-ci.org, but not on forks on travis-ci.com (!?!)

Failing: https://travis-ci.org/ankidroid/Anki-Android/jobs/607187558

Just fine (same tag): https://travis-ci.com/mikehardy/Anki-Android/jobs/252864960

Just fine (different dev, same tag): https://travis-ci.com/Overflow0xFFFF/Anki-Android/jobs/252834126

**Comment by mikehardy** [comment]
This is sort of a catch-all issue for CI flakiness at the moment.

I just noticed a new one, the jabba install failed on an I/O timeout from github fetching the JDK for install. Similar to sdkmanager I suppose this network fetch needs a retry as well?

**Comment by mikehardy** [comment]
This one is close and still needed - Travis still false-negatives frequently

 we do travis_retry on some things now but not all. The jabba failures are the most frequent things now, perhaps pulling jabba into a separate script with careful testing for success at every step (valid jabba gzip, valid jdk fetch, valid java at the end) and reporting that back up for travis_retry to work would do it

**Comment by mikehardy** [comment]
Here's one failure mode that is regular enough to combat. The emulator starts but then is unresponsive to adb commands, and then the build hangs and is terminated after 10mins. Perhaps after wait_for_emulator, wait a bit more, then try an adb command and if unsuccessful return error code. On error code kill the emulator. Put that in a loop with 2 retries perhaps

<details>

```
Waiting for emulator to start

Waiting for emulator to start

Waiting for emulator to start

Waiting for emulator to start

Waiting for emulator to start

Waiting for emulator to start

Waiting for emulator to start

emulator: onGuestSendCommand: [0x72720d0] Adb connected, start proxing data

emulator: _hwFingerprint_connect: connect finger print listen is called

emulator: got message from guest system fingerprint HAL

emulator: INFO: boot completed

emulator: INFO: boot time 63787 ms

emulator: Increasing screen off timeout, logcat buffer size to 2M.

emulator: Revoking microphone permissions for Google App.

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell settings put system screen_off_timeout 2147483647 |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell logcat -G 2M |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell logcat -G 2M |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell logcat -G 2M |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell logcat -G 2M |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell logcat -G 2M |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell pm revoke com.google.android.googlequicksearchbox android.permission.RECORD_AUDIO |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell pm revoke com.google.android.googlequicksearchbox android.permission.RECORD_AUDIO |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell pm revoke com.google.android.googlequicksearchbox android.permission.RECORD_AUDIO |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell pm revoke com.google.android.googlequicksearchbox android.permission.RECORD_AUDIO |

emulator: VERBOSE: System.cpp:1487: Timed out with running command |/home/travis/android-sdk/platform-tools/adb -e shell pm revoke com.google.android.googlequicksearchbox android.permission.RECORD_AUDIO |

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received

The build has been terminated
```
</details>

**Comment by mikehardy** [comment]
Peeled the one remaining item out into a new issue, and I'm going to close this epic. Travis is in great shape vs where it was, we burn maybe 50,000 CI minutes a month with only one flake every couple days now.
